lambdas/tests/unit/services/test_document_service.py

```diff
@@ -30,59 +31,74 @@ def nhs_number():
     return "9000000009"
 
 
-def test_returns_list_of_documents_when_results_are_returned(nhs_number):
-    expected_table = "expected_table_name"
-    os.environ["LLOYD_GEORGE_DYNAMODB_NAME"] = expected_table
-    os.environ["DOCUMENT_STORE_DYNAMODB_NAME"] = "no-table"
+def test_returns_list_of_lg_document_references_when_results_are_returned(
+    set_env, mocker
+):
+    mocker.patch("boto3.resource")
+    service = DocumentService()
 
-    with patch.object(boto3, "resource", return_value=MagicMock()) as mock_dynamo:
-        mock_table = MagicMock()
-        mock_dynamo.return_value.Table.return_value = mock_table
-        mock_table.query.return_value = MOCK_SEARCH_RESPONSE
-        result = DocumentService().fetch_available_document_references_by_type(
-            nhs_number, "LG"
-        )
+    mock_dynamo_table = mocker.patch.object(service.dynamodb, "Table")
+    mock_dynamo_table.return_value.query.return_value = MOCK_SEARCH_RESPONSE
 
-        mock_dynamo.return_value.Table.assert_called_with(expected_table)
+    result = service.fetch_available_document_references_by_type(TEST_NHS_NUMBER, "LG")
 
-        assert len(result) == 3
-        assert type(result[0]) == DocumentReference
+    mock_dynamo_table.assert_called_with(MOCK_LG_TABLE_NAME)
 
+    assert len(result) == 3
+    assert type(result[0]) == DocumentReference
```

Comment:
for this line, my linter suggested to change to `assert type(result[0]) is DocumentReference` or `assert isinstance(result[0], DocumentReference)` 
Personally I think in our use case using == to compare isn't a problem so up to you :)

Reply:
No I agree, cleaner test for equality